apps/ui/src/views/Space/Pro.vue

```diff
@@ -129,37 +210,43 @@ const features = computed<Feature[]>(() => {
 
     <div class="mx-4 space-y-4 flex flex-col items-center">
       <div class="max-w-[480px] w-full space-y-3">
-        <a
-          class="border rounded-lg px-4 py-3 flex gap-2 justify-between"
-          :class="subscriptionLength === 'monthly' && 'border-skin-link'"
-          @click="subscriptionLength = 'monthly'"
-        >
-          <h3 class="flex-1">Pay monthly</h3>
-          <div class="flex items-center space-x-1">
-            <h2>$600</h2>
-            <span class="text-sm text-skin-text">/ month</span>
-          </div>
-        </a>
-        <a
-          class="border rounded-lg px-4 py-3 flex gap-2 justify-between"
-          :class="subscriptionLength === 'yearly' && 'border-skin-link'"
-          @click="subscriptionLength = 'yearly'"
+        <button
+          v-for="plan in Object.keys(PRO_MONTHLY_PRICES)"
+          :key="plan"
+          :class="[
+            'border rounded-lg px-4 py-3 flex gap-2 justify-between w-full',
+            { 'border-skin-link': subscriptionLength === plan }
+          ]"
+          @click="subscriptionLength = plan as SubscriptionLength"
         >
           <div class="flex flex-1 items-center gap-x-2 flex-wrap">
-            <h3>Pay yearly</h3>
-            <div>
+            <h3 class="text-start">Pay {{ plan }}</h3>
+            <div v-if="plan === 'yearly'">
               <div class="bg-skin-border text-sm rounded-full px-2">
-                Save $1,200
+                Save ${{
+                  _n(
+                    (
+                      (PRO_MONTHLY_PRICES.monthly - PRO_MONTHLY_PRICES.yearly) *
+                      12
+                    ).toFixed(0)
+                  )
+                }}
               </div>
             </div>
           </div>
-          <div class="flex items-center space-x-1 flex-wrap">
-            <h2>$500</h2>
+          <div class="flex items-center justify-end space-x-1 flex-wrap">
+            <h2>${{ PRO_MONTHLY_PRICES[plan] }}</h2>
             <span class="text-sm text-skin-text">/ month</span>
           </div>
-        </a>
+        </button>
       </div>
-      <UiButton class="primary">Upgrade {{ space.name }}</UiButton>
+      <UiButton
+        class="primary"
+        :disabled="space.network !== metadataNetwork"
+        @click="handleTurboClick"
+      >
+        {{ space.turbo ? 'Extend' : 'Upgrade' }} {{ space.name }}
+      </UiButton>
```

Comment:
Would be nice to show current expiry somewhere here

Reply:
@bonustrack

Reply:
We will make it into another page later